### PR TITLE
Update machine-controller to v1.0.7 to allow initialization

### PR DESCRIPTION
**Pull request by alvaroaleman**
**What this PR does / why we need it**:

Updates the machine-controller to v1.0.7, this is required to allow the ipam-controller to work based off of annotations instead of initializers.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Documentation**:

**Release note**:

```release-note
NONE
```

/assign @mrIncompetent

### CODE REVIEW

File: api/Gopkg.toml [hunk 1]
```diff
@@ -121,7 +121,7 @@ required = [
 
 [[override]]
   name = "github.com/kubermatic/machine-controller"
-  version = "v1.0.6"
+  version = "v1.0.7"
```
Comment (mrIncompetent) [inline api/Gopkg.toml:124]: Even though we changed that file nothing in the vendor dir changed?

Comment (alvaroaleman) [reply api/Gopkg.toml:124]: Apparently, yes

**Comment by alvaroaleman** [comment]
Actually I will close this. It was a terrible idea to define a const that is bascially an api in the same place were all the business logic is. Also it contradicts the idea of not being dependent on the cluster api typed clients.

/close

**Comment by alvaroaleman** [comment]
Actually, even with the const defined in the wrong package we need the functionality for #2940 so reopening. Sorry for the noise :sa: 

/reopen

**Pull request merged**